src/porepy/params/bc.py

```diff
@@ -57,7 +59,13 @@ class BoundaryCondition(AbstractBoundaryCondition):
             face i has been assigned a Neumann condition.
         is_rob (np.ndarray, boolean, size g.num_faces): Element i is true if
             face i has been assigned a Robin condition.
-
+        is_per (np.ndarray, boolean, size g.num_faces): Element i is true if
+            face i has been assigned a periodic boundary condition. Note that
+            periodic boundary conditions are non-standard and might not work
+            for all discretizations. See also attribute per_map
+        per_map (np.ndarray, int, 2 x # periodic faces): Defines the periodic
```

Comment:
Is there an assumption that the paired faces matches geometrically (perfect overlap)?

Reply:
Both yes and no. The BoundaryCondition class does not know anything about the geometry of the faces, it only stores which faces should be coupled. So it could couple two faces that do not matches. It does have a limitation that it can only couple exactly two faces. 

But the discretization assumes that the faces are matching. This is the only thing that makes sense for the TPFA discretization if the periodic faces are supposed to be treated exactly the same as an internal face (the face between two cells is assumed to be unique).

I think adding a mortar variable is the way to go if we want non-matching faces. In that case, we would have to rethink the per_map in the BoundaryCondition class.

Reply:
Let's leave it like this for now, then.